docs/src/format.md

```diff
@@ -0,0 +1,441 @@
+
+```hdf5
+parameters.hdf5
+└───layer1 (group)
+│   ├── arrayId{1}
+│   └── arrayId{2}
+└───layer2 (group)
+    ├── arrayId{1}
+    └── arrayId{2}
+```
+
+Here, `arrayId` depends on the naming convention for parameters in each layer. For example, a PyTorch `linear` layer typically has arrays named `weight` and (optionally) `bias`. While these names are common in many layers, the actual `arrayId` depends on the layer type and the specific neural network library.
+
+Because parameters are stored in HDF5, they are saved as arrays. The indexing convention and naming therefore depends on the model library:
+
+- For neural network models in the PEtab SciML YAML format, indexing follows PyTorch conventions. Usually, users do not need to handle these details directly, as PEtab SciML tools manage them automatically. This also means that `arrayId` follows PyTorch naming convention.
+- For neural networks provided by another library, the indexing and `arrayId` naming follow that library’s conventions.
+
+### [Neural Network Input](@id hdf5_input_structure)
+
+When network input is provided as an array via an HDF5 file (see the mapping table below), its format should be:
+
+```hdf5
+input.hdf5
+└───input (group)
+    └─── input_array
+```
+
+As with [parameters](@ref hdf5_ps_structure), the indexing depends on the neural network library:
+
+- For neural network models in the PEtab SciML format, indexing follows PyTorch conventions. For example, if the first layer is `Conv2d`, the input should be in `(C, W, H)` format, with data stored in row-major order. In general, the input should be structured to be directly compatible with PyTorch.
+- For neural networks provided by another library, the indexing and ordering follow the conventions of that library.
+
+!!! tip "For developers: Respect memory order"
+    Tools supporting the SciML extension should, for computational efficiency, reorder input data and potential layer parameter arrays to match the memory ordering of the target language. For example, PEtab.jl converts input data to column-major order, with Julia indexing.
+
+TODO: We will fix condition specific input in the YAML file later.
+
+## [Mapping Table](@id mapping_table)
+
+Each neural network is assigned an Id in the PEtab SciML extension section of the PEtab problem [YAML](@ref YAML_file) file. To avoid confusion about what a neural network Id refers to (e.g., parameters, inputs, outputs), a neural network Id is not a valid PEtab identifier. Therefore, every neural network input, parameter, and output that will be referenced in the PEtab problem must be mapped to PEtab identifiers problem via the mapping table. Repeating parts of the PEtab standard, the key columns in the mapping table are:

```

Comment:
```suggestion
Each neural network is assigned an Id in the PEtab SciML extension section of the PEtab problem [YAML](@ref YAML_file) file. To avoid confusion about what a neural network Id refers to (e.g., parameters, inputs, outputs), a neural network Id is not a valid PEtab identifier. Therefore, every neural network input, parameter, and output that will be referenced in the PEtab problem must be mapped to PEtab identifiers via the mapping table. Repeating parts of the PEtab standard, the key columns in the mapping table are:

```